File: framework/languages/java/samples/java/DeliveryDispatch/Server/CustomerGateway/src/main/java/systems/zlink/samples/deliverydispatch/server/customergateway/spots/handlers/DeliveryStatusUpdatedHandler.java
package systems.zlink.samples.deliverydispatch.server.customergateway.spots.handlers;

import systems.zlink.framework.spots.ZLinkEntrySpotActorSendHandler;
import systems.zlink.framework.spots.ZLinkSpotActorSendContext;
import systems.zlink.samples.deliverydispatch.server.customergateway.CustomerActor;
import systems.zlink.samples.deliverydispatch.server.customergateway.spots.CustomerEntrySpot;
import systems.zlink.samples.deliverydispatch.shared.contracts.Messages;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

public final class DeliveryStatusUpdatedHandler
    implements ZLinkEntrySpotActorSendHandler<
        CustomerEntrySpot,
        CustomerActor,
        Messages.DeliveryStatusUpdatedMsg> {
    @Override
    public CompletionStage<Void> handle(
        CustomerEntrySpot entrySpot,
        CustomerActor actor,
        ZLinkSpotActorSendContext context,
        Messages.DeliveryStatusUpdatedMsg message) {
        actor.push(new Messages.DeliveryStatusNotify(
            message.deliveryId(),
            message.status(),
            message.courierId(),
            message.occurredAt()));
        return CompletableFuture.completedFuture(null);
    }
}

File: framework/languages/java/samples/kotlin/DeliveryDispatch/Server/CustomerGateway/src/main/kotlin/systems/zlink/samples/kotlin/deliverydispatch/server/customergateway/spots/handlers/DeliveryStatusUpdatedHandler.kt
package systems.zlink.samples.kotlin.deliverydispatch.server.customergateway.spots.handlers

import systems.zlink.framework.CancellationToken
import systems.zlink.framework.spots.ZLinkEntrySpotActorSendHandler
import systems.zlink.framework.spots.ZLinkSpotActorSendContext
import systems.zlink.samples.kotlin.deliverydispatch.server.customergateway.CustomerActor
import systems.zlink.samples.kotlin.deliverydispatch.server.customergateway.spots.CustomerEntrySpot
import systems.zlink.samples.kotlin.deliverydispatch.shared.contracts.DeliveryStatusNotify
import systems.zlink.samples.kotlin.deliverydispatch.shared.contracts.DeliveryStatusUpdatedMsg

class DeliveryStatusUpdatedHandler : ZLinkEntrySpotActorSendHandler<
    CustomerEntrySpot,
    CustomerActor,
    DeliveryStatusUpdatedMsg,
    > {
    override fun handle(
        entrySpot: CustomerEntrySpot,
        actor: CustomerActor,
        context: ZLinkSpotActorSendContext,
        message: DeliveryStatusUpdatedMsg,
        cancellationToken: CancellationToken,
    ) {
        actor.push(
            DeliveryStatusNotify(
                deliveryId = message.deliveryId,
                status = message.status,
                courierId = message.courierId,
                occurredAt = message.occurredAt,
            ),
        )
    }
}
